packages/vscode-integration-webview/src/glsp-starter.ts

```diff
@@ -25,22 +27,29 @@ import {
     TYPES
 } from '@eclipse-glsp/client';
 import { Container } from 'inversify';
-import {
-    SprottyDiagramIdentifier,
-    SprottyStarter,
-    VscodeDiagramServer,
-    VscodeDiagramWidget,
-    VscodeDiagramWidgetFactory
-} from 'sprotty-vscode-webview';
-import { VsCodeApi } from 'sprotty-vscode-webview/lib/services';
 import { CopyPasteHandlerProvider } from './copy-paste-handler-provider';
-import { GLSPDiagramIdentifier, isDiagramIdentifier } from './diagram-identifer';
+import { GLSPDiagramIdentifier, isDiagramIdentifier, WebviewReadyMessage } from './diagram-identifier';
 import { GLSPVscodeExtensionActionHandler } from './extension-action-handler';
-import { GLSPVscodeDiagramWidget } from './glsp-vscode-diagram-widget';
+import { GLSPDiagramWidget, GLSPDiagramWidgetFactory } from './glsp-diagram-widget';
 import { GLSPVscodeDiagramServer } from './glsp-vscode-diagramserver';
+import { GLSPStarterServices, VsCodeApi } from './services';
+
+declare function acquireVsCodeApi(): VsCodeApi;
+
+export abstract class GLSPStarter {
+    protected container?: Container;
+    protected vscodeApi: VsCodeApi;
+
+    constructor(services: GLSPStarterServices = {}) {
+        this.vscodeApi = services.vscodeApi ?? acquireVsCodeApi();
+        this.sendReadyMessage();
+        this.acceptDiagramIdentifier();
+    }
 
-export abstract class GLSPStarter extends SprottyStarter {
-    protected override acceptDiagramIdentifier(): void {
+    protected sendReadyMessage(): void {
+        this.vscodeApi.postMessage({ readyMessage: 'Sprotty Webview ready' } as WebviewReadyMessage);
```

Comment:
I see that we are also dropping `vscode-messenger-webview` and its messenger API. It is fine by me but I just wanted to make that very explicit.

Reply:
No we are not. Previously we were dependent on sprotty-vscode-webview in version 0.3.x. The `vscode-messenger-webview` dependency was introduced in later versions. 
If you have a look at the sprotty-starter class of 3.0 you can see that the `sendReadyMessage()` implementation is exactly the same:
https://github.com/eclipse-sprotty/sprotty-vscode/blob/7980fca22f4504d4431bf971ffc7fd27be75a180/sprotty-vscode-webview/src/sprotty-starter.ts#L39

FYI, the switch to `vscode-messenger-webview` is one of the reasons why we decided to drop the dependency to sprotty-vscode entirely. It introduces yet another abstraction and interdependency and provides little to no benefit for us.